yoti_python_sdk/tests/doc_scan/session/create/test_notification_config.py

```diff
@@ -104,6 +105,41 @@ def test_should_store_unique_topics(self):
 
         assert len(result.topics) == 1
 
+    def test_should_not_build_with_invalid_auth_type(self):
+        with self.assertRaises(ValueError) as context:
+            NotificationConfigBuilder()\
+                .with_auth_token(self.SOME_AUTH_TOKEN)\
+                .with_endpoint(self.SOME_ENDPOINT)\
+                .with_topic(self.SOME_TOPIC)\
+                .with_auth_type('INVALID')\
+                .build()
+
+    def test_should_build_with_basic_type(self):
+        result = (
+            NotificationConfigBuilder()
+            .with_auth_token(self.SOME_AUTH_TOKEN)
+            .with_endpoint(self.SOME_ENDPOINT)
+            .with_topic(self.SOME_TOPIC)
+            .with_auth_type(constants.DocScanAuthType.BASIC.value)
+            .build()
+        )
+
+        assert isinstance(result, NotificationConfig)
+        assert result.auth_type == constants.DocScanAuthType.BASIC.value
+
+    def test_should_build_with_bearer_type(self):
+        result = (
+            NotificationConfigBuilder()
+            .with_auth_token(self.SOME_AUTH_TOKEN)
+            .with_endpoint(self.SOME_ENDPOINT)
+            .with_topic(self.SOME_TOPIC)
+            .with_auth_type(constants.DocScanAuthType.BEARER.value)
+            .build()
+        )
+
+        assert isinstance(result, NotificationConfig)
+        assert result.auth_type == constants.DocScanAuthType.BEARER.value
```

Comment:
Do you mind to change change these two tests into one parameterized test ?

You could simplify the two tests which test the `auth_type` into one test with a parameterized test, the only difference between each test is the input into the builder functions. Using a parameterized test will prevent duplication if we add any more tests which test the `auth_type` too. 

I normally use `pytest` which has a decorator for parameterized tests but for `unittest` you should be able to do something similar. `parameterized.expand` will internally create the two tests for you.

```
from parameterized import parameterized

class NotificationConfigTest(unittest.TestCase):
      @parameterized.expand([
       ('name_of_test1', 'auth_token1', 'endpoint1', 'topic1', 'auth_type1', 'expected_config_instance1', 'expected_auth_type1'),
       ('name_of_test2', 'auth_token2', 'endpoint2', 'topic2', 'auth_type2', 'expected_config_instance2', 'expected_auth_type2'),
   ])
   
   def test_notification_config_is_constructed_with_correct_auth_type(self, auth_token, endpoint, topic, auth_type,
                                                                      expected_config_instance, expected_auth_type):
        result = (
            NotificationConfigBuilder()
            .with_auth_token(auth_token)
            .with_endpoint(endpoint)
            .with_topic(topic)
            .with_auth_type(auth_type)
            .build()
        )

        assert isinstance(expected_config_instance, NotificationConfig)
        assert result.auth_type == expected_auth_type
```

Reply:
I will recommend to stay with what I pushed.
1. it's much more easier to debug with separate tests
2. in future behave for config can be pretty diffrent depends on auth_type, it will be much easier to modify separate tests
3. in my opinion it's more readable when someone will manage code
4. my expierence with parameterized is not so positive. I mean ok, why not. Seems ok but like above and it's not common pattern.

Reply:
That's fine if you prefer to keep it the way it is:+1: 

Just to answer some of your points.

1. Its pretty easy to debug a parameterized test because when there's a test failure the test case and the inputs which caused that test to fail are logged making it easy to track down the error.

2. For modifying separate tests if the config behaves different, I think it would be easier to just update the params in the paramerterized test and the test logic once rarther than having to go back and update multiple test cases.

And I think its quite a common pattern too, its used a bit in this repo see [here](https://github.com/getyoti/yoti-python-sdk/search?p=1&q=parametrize).